examples/plot_metrics_with_error.py

```diff
@@ -0,0 +1,162 @@
+# https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Normal_approximation_interval
+#
+# We aim to create a 95% confidence interval, so we use a `z_score` of 1.959964
+z_score = 1.959964
+digits_of_precision = 4
+
+
+def wilson(p, n, digits=digits_of_precision, z=z_score):
+    """ Returns lower and upper bound """
+    denominator = 1 + z**2/n
+    centre_adjusted_probability = p + z*z / (2*n)
+    adjusted_standard_deviation = np.sqrt((p*(1 - p) + z*z / (4*n)))/np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z*adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z*adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+def compute_error_metric(metric_value, sample_size, z_score):
+    return z_score*np.sqrt(metric_value*(1.0-metric_value))/np.sqrt(sample_size)
+
+
+def recall_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    tn, fp, fn, tp = confusion_matrix(y_true, y_pred).ravel()
+    bounds = wilson(tp/(tp+fn), tp + fn, digits_of_precision, z_score)
+    return bounds
+
+def recall_normal_err(y_t, y_p):
+    assert len(y_t) == len(y_p)
+    tn, fp, fn, tp = confusion_matrix(y_t, y_p).ravel()
+    error = compute_error_metric(tp/(tp+fn), tp + fn, z_score=z_score)
+    return (error, error)
+
+def accuracy_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    bounds = wilson(score, len(y_true), digits_of_precision, z_score)
+    return bounds
+
+def accuracy_normal_err(y_t, y_p):
+    assert len(y_t) == len(y_p)
+    score = accuracy_score(y_true, y_pred)
+    error = compute_error_metric(score, len(y_true), z_score)
+    return (error, error)
+
+
+# %%
+# Classifier
+# ----------
+# Now we use a Decision Tree to make predictions
+#
+
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+classifier.fit(X, y_true)
+y_pred = classifier.predict(X)
+
+# %%
+# MetricFrame
+# -----------
+# Now we create a MetricFrame to generate the Recall wilson bounds and Recall normal error
+#
+
+# Analyze metrics using MetricFrame
+metrics_dict = {
+    'Recall': recall_score,
+    'Recall Bounds': recall_wilson,
+    'Recall Error': recall_normal_err,
+    'Accuracy': accuracy_score,
+    'Accuracy Bounds': accuracy_wilson,
+    'Accuracy Error': accuracy_normal_err,
+}
+metric_frame = MetricFrame(metrics_dict, y_true, y_pred, sensitive_features=sex)
+
+# %%
+# plot metrics with (symmetric) error bars
+MetricFrame.plot_metric_frame("bar", metric_frame, metrics=['Recall', 'Accuracy'], error_bars=['Recall Error', 'Accuracy Error'])
+MetricFrame.plot_metric_frame("bar", metric_frame, metrics='Recall', error_bars='Recall Error')
+
+# %%
+# plot metrics with confidence intervals (possibly asymmetric)
+MetricFrame.plot_metric_frame("bar", metric_frame, metrics=['Recall', 'Accuracy'], conf_intervals=['Recall Bounds', 'Accuracy Bounds'])
```

Comment:
For wider discussion: Should this be a static method on `MetricFrame` or just a standalone function? Since it doesn't require access to the internals, I'd be inclined to go with the latter.

Reply:
Agreed. Let's do a standalone function `metrics.plot_metric_frame()` that nicely matches `postprocessing.plot_threshold_optimizer()`.